test/unit/org/apache/cassandra/contraints/ScalarSatisfiabilityTest.java

```diff
@@ -0,0 +1,159 @@
+
+    @Test
+    public void testSatisfiability() throws Throwable
+    {
+        for (Operator op1 : SUPPORTED_OPERATORS)
+        {
+            for (Operator op2 : SUPPORTED_OPERATORS)
+            {
+                if (op1 == op2)
+                {
+                    if (op1 == NEQ)
+                    {
+                        // a_column != 0 and a_column != 10 -> valid
+                        check(op1, 0, op2, 10, null);
+                        // does not make sense to check twice
+                        // check a_column != 0 and a_column != 0
+                        check(op1, 0, op2, 0, String.format("There are duplicate constraint definitions on column '%s'.", columnIdentifier));
+                    }
+                    else
+                        check(op1, 0, op2, 10, "There are duplicate constraint definitions on column 'a_column': [a_column " + op1 + ']');
+                }
+                else if ((op1 == GT && op2 == GTE) ||
+                         (op1 == GTE && op2 == GT) ||
+                         (op1 == LT && op2 == LTE) ||
+                         (op1 == LTE && op2 == LT) ||
+                         (op1 == EQ || op2 == EQ))
+                {
+                    check(op1, 0, op2, 10, "Scalar constraints combination is not supported: a_column " + op1 + " 0, a_column " + op2 + " 10");
+                }
+                else if ((op1 == LTE && op2 == GT) ||
+                         (op1 == LT && op2 == GT) ||
+                         (op1 == LTE && op2 == GTE) ||
+                         (op1 == LT && op2 == GTE))
+                {
+                    check(op1, 0, op2, 10, "Scalar constraints are not satisfiable: a_column " + op1 + " 0, a_column " + op2 + " 10");
+                }
+                else if (!(op1 == NEQ || op2 == NEQ))
+                {
+                    check(op1, 0, op2, 10, null);
+                }
+                else
+                {
+                    // this is valid
+                    // a_column < 0, a_column != 10
+                }
+            }
+        }
+    }
+
+    @Test
+    public void testNumberOfScalarConstraints()
+    {
+        // one
+        new ColumnConstraints(of(get(LT, 5))).checkInvalidConstraintsCombinations(columnIdentifier, IntegerType.instance);
+
+        // two
+        new ColumnConstraints(of(get(LT, 5), get(GT, 0))).checkInvalidConstraintsCombinations(columnIdentifier, IntegerType.instance);
+
+        // three - invalid
+
+        assertThatThrownBy(() -> new ColumnConstraints(of(get(LT, 5), get(GT, 0), get(GTE, 0)))
+                                 .checkInvalidConstraintsCombinations(columnIdentifier, IntegerType.instance))
+        .hasMessage("There can not be more than 2 scalar constraints (not including non-equal relations) on a column 'a_column' but you have specified 3");
+
+        // valid
+        new ColumnConstraints(of(get(LT, 5), get(GT, 0), get(NEQ, 3)))
```

Comment:
Add a 
```
new ColumnConstraints(of(get(EQ, 5), get(EQ, 0)))
        .checkInvalidConstraintsCombinations(columnIdentifier, IntegerType.instance);
```

case

Reply:
@bbotella that is covered already, this is a case of duplicity.